draft-ietf-quic-transport.md

```diff
@@ -1207,41 +1208,53 @@ be routed back to the endpoint and identified by the endpoint upon receipt.
-series of connection IDs over the lifetime of a connection. The series of
-connection IDs issued by an endpoint is ordered, with the final connection ID
-selected during the handshake coming first.  Additional connection IDs are
-provided using the NEW_CONNECTION_ID frame ({{frame-new-connection-id}}), each
-with a specified sequence number.  The series of connection IDs issued SHOULD be
-contiguous, but might not appear to be upon receipt due to reordering or loss.
-
-Each connection ID MUST be used on only one local address. When packets are sent
-for the first time on a new local address, a new connection ID MUST be used with
-a higher sequence number than any connection ID previously used on any local
-address.  At any time, an endpoint MAY change to a new connection ID on a local
-address already in use.
-
-An endpoint MUST NOT send packets with a connection ID which has a lower
-sequence number than the highest sequence number of any connection ID ever sent
-or received on that local address.  This ensures that when an endpoint migrates
+An endpoint whose peer has selected a non-zero-length connection ID will
+potentially use multiple connection IDs over the lifetime of a connection.
+These connection IDs are supplied by the peer using the NEW_CONNECTION_ID frame
+({{frame-new-connection-id}}).
+
+An endpoint considers the entire set of connection IDs that its peer has issued
+to be valid for use at any time. When the peer issues a new connection ID, it
+MUST accept packets using this connection ID for the duration of the connection
+or until the peer invalidates the connection ID via a CONNECTION_ID_FINISHED
+frame.
+
+Endpoints may choose to stop using a given connection ID to send packets at any
+time and signals this to the issuing endpoint via a CONNECTION_ID_FINISHED
+frame.  This frame indicates the connection ID that is no longer in use and
+serves as a request for the peer to issue additional connection IDs via a
+NEW_CONNECTION_ID frame.
+
+
+Each connection ID MUST be used on only one local address.  At any time, an
+endpoint MAY change to a new connection ID on a local address already in use. An
+endpoint that migrates to a new local address may wish to retire all connection
+IDs that were used on the previous address using the CONNECTION_ID_FINISHED
+frame. This allows the server to provide new connection IDs which are valid on
+any address until first use.
+
+When a receiver of a packet notices that its peer is now using a previously
+unused connection ID, it may choose to supply its peer with a new connection ID
+using a NEW_CONNECTION_ID frame to reduce the possibility of its peer running
+out of available connection IDs.
+
+If an endpoint receives a packet with a previously unused connection ID, it
+SHOULD use a different connection ID from its set of valid connection IDs when
+sending its responses.  This can help to ensure that when an endpoint migrates
```

Comment:
Might want to mention that this is only if the other side isn't using 0-length CIDs

Reply:
Done